gen/http/httpclient/generator.go

```diff
@@ -373,7 +374,15 @@ func (g *Generator) Generate(pkgInfo *generator.PkgInfo, ifaceData *ifacetool.Da
 				return param.Name
 			},
 			"returnErr": func(params []*ifacetool.Param) string {
-				emptyValue := func(typ string) string {
+				emptyValue := func(param *ifacetool.Param) string {
+					t := param.Type.Underlying()
+					typ := param.TypeString
+
+					_, ok := t.(*types.Interface)
+					if ok {
+						return "nil"
+					}
+
 					switch typ {
```

Comment:
I think the following code also needs to change because `typ` is not a string any more.